ui/src/components/search.tsx

```diff
@@ -77,23 +79,23 @@ export class Search extends Component<any, SearchState> {
     },
   };
 
-  getSearchQueryFromProps(props: any): string {
+  static getSearchQueryFromProps(props: any): string {
```

Comment:
I'm not sure if we actually gain anything by making these static, but I have no problem with it.

Reply:
They need to be static to be accessible from `getDerivedState` I think?  I got errors otherwise, but the User component doesnt have this issue and I'm not sure why

Reply:
It basically made it above so you could call `Search.staticFunction` rather than `this.method` . Its no prob cause it works both ways.